src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/kafka/InfoDataDeserializer.java

```diff
@@ -15,18 +15,40 @@
 
 package org.openkilda.wfm.kafka;
 
+import static java.lang.String.format;
+
 import org.openkilda.messaging.info.InfoData;
 import org.openkilda.wfm.topology.utils.SerializationUtils;
 
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.common.serialization.Deserializer;
 
 import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Map;
 
 @Slf4j
-public class InfoDataDeserializer extends Deserializer<InfoData> {
+public class InfoDataDeserializer implements Deserializer<InfoData> {
+
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        // No-op
+    }
+
+    @Override
+    public InfoData deserialize(String topic, byte[] data) {
+        try {
+            return SerializationUtils.MAPPER.readValue(data, InfoData.class);
+        } catch (IOException e) {
+            log.error(format("Failed to deserialize data: %s from topic %s",
+                    StringUtils.toEncodedString(data, Charset.defaultCharset()), topic), e);
+            return InfoData.CORRUPTED_INFO_DATA;
```

Comment:
are there way not to return anything in this case?

Reply:
Reworked to return InfoData with error details.